packages/react-devtools-shared/src/utils.js

```diff
@@ -1017,7 +1017,6 @@ export function backendToFrontendSerializedElementMapper(
   };
 }
 
-// This is a hacky one to just support this exact case.
 export function normalizeUrl(url: string): string {
-  return url.replace('/./', '/');
+  return new URL(url).toString();
```

Comment:
Will this normalize URLs with `webpack-internal` protocol?

Reply:
```js
new URL('webpack-internal://project/./src/index.ts').toString()
"webpack-internal://project/./src/index.ts" 
```

No. Should it?

Reply:
> ```js
> new URL('webpack-internal://project/./src/index.ts').toString()
> "webpack-internal://project/./src/index.ts" 
> ```
> 
> No. Should it?

Yes, I think this case was an explicit one for Next, if I am not mistaken.

The urls in source map annotation can be nonnormalized, but Chrome will actually normalize urls for every resource in inspected window, [this is where we call it](https://github.com/facebook/react/blob/3ac551e855f9bec3161da2fc8787958aa62113db/packages/react-devtools-extensions/src/main/fetchFileWithCaching.js#L116-L118).

Technically, we can remove the `normalizeUrl` function and do a bit more work in `fetchFileWithCaching.js`, so we can compare normalized urls with nonnormalized ones.